src/bundle/Controller/LocationController.php

```diff
@@ -41,10 +46,74 @@ public function actionsAction(
                 '_ez_content_view',
                 [
                     'contentId' => $content->id,
-                    'layout' => 'true',
-                    'viewType' => 'full',
                 ]
             )
         );
     }
+
+    public function visibilityAction(
+        Request $request,
+        LocationService $locationService,
+        UiFormFactory $formFactory,
+        TranslatorInterface $translator,
+        NotificationBuilder $notificationBuilder
+    ) {
+        try {
+            $visibilityForm = $formFactory->createLocationsVisibilityForm();
+            $visibilityForm->handleRequest($request);
+
+            if (!$visibilityForm->isValid()) {
+                throw new \Exception('Invalid form submission.');
+            }
+
+            $visibility = $visibilityForm->get('visibility')->getData();
+            $locationId = $visibilityForm->get('locationId')->getData();
+
+            $location = $locationService->loadLocation($locationId);
+
+            if ($visibility) {
+                $locationService->unhideLocation($location);
+                /** @Desc("The Location #%id% is now visible") */
+                $message = $translator->trans(
+                    'locationview.locations.notification.visible', ['id' => $location->id], 'locationview'
```

Comment:
It may be worth setting the translation domain as a constant and use that instead to stop it being defined in multiple places?